compiler-core/src/ast.rs

```diff
@@ -255,6 +255,7 @@ pub struct RecordConstructor<T> {
     pub location: SrcSpan,
     pub name_location: SrcSpan,
     pub name: EcoString,
+    pub type_name: EcoString,
```

Comment:
Remove this please 🙏 All constructors share the same type, and that type is stored at the level above, so this is duplicating the information already there. Instead pass is down into the function as needed.